libfwupdplugin/fu-common.c

```diff
@@ -3189,3 +3189,30 @@ fu_common_uri_get_scheme (const gchar *uri)
 		return NULL;
 	return g_utf8_strdown (uri, tmp - uri);
 }
+
+/**
+ * fu_common_align_up:
+ * @value: value to align
+ * @alignment: align to this power of 2
+ *
+ * Align a value to a power of 2 boundary, where @alignment is the bit position
+ * to align to. If @alignment is zero then @value is always returned unchanged.
+ *
+ * Returns: aligned value, which will be the same as @value if already aligned
```

Comment:
I think you might need a check that you aren't getting bigger than `G_MAXSIZE` as a result of this rounding up.

Reply:
Good idea. Fixed.